File: vb/word_reverser/word_reverser_11.vb
' This program is called Word Reverser.
' The purpose of this program is to reverse the order of characters in each word of a given sentence.
' It uses regular expressions to identify words and reverse them.
' This program is designed to help programmers understand how to use regular expressions in Visual Basic.

Imports System.Text.RegularExpressions

Module WordReverser

    Sub Main()
        ' Define the input sentence
        Dim inputSentence As String = "Hello world from Visual Basic"

        ' Define the regular expression pattern to match words
        Dim pattern As String = "\b\w+\b"

        ' Use Regex.Replace to find matches and reverse each word
        Dim reversedSentence As String = Regex.Replace(inputSentence, pattern, Function(m) ReverseString(m.Value))

        ' Print the reversed sentence
        Console.WriteLine(reversedSentence)
    End Sub

    ' Function to reverse a given string
    Function ReverseString(ByVal input As String) As String
        ' Convert the input string to a character array
        Dim charArray() As Char = input.ToCharArray()

        ' Reverse the character array
        Array.Reverse(charArray)

        ' Convert the character array back to a string and return it
        Return New String(charArray)
    End Function

End Module

File: swift/word_reverser/word_reverser_11.swift
// This program is called "Word Reverser". It is designed to reverse the words in a given string.
// The program uses regular expressions to identify words and reverse them.
// The purpose of this program is to help programmers practice their skills in Swift and regular expressions.

import Foundation

// Function to reverse words in a given string
func reverseWords(in input: String) -> String {
    // Regular expression pattern to match words
    let pattern = "\\b\\w+\\b"
    
    // Create a regular expression object
    guard let regex = try? NSRegularExpression(pattern: pattern, options: []) else {
        return input
    }
    
    // Find matches in the input string
    let matches = regex.matches(in: input, options: [], range: NSRange(location: 0, length: input.utf16.count))
    
    // Convert input string to NSString for easier manipulation
    let nsString = input as NSString
    
    // Array to hold reversed words
    var reversedWords = [String]()
    
    // Iterate over matches and reverse each word
    for match in matches {
        let wordRange = match.range
        let word = nsString.substring(with: wordRange)
        let reversedWord = String(word.reversed())
        reversedWords.append(reversedWord)
    }
    
    // Join reversed words with spaces
    let result = reversedWords.joined(separator: " ")
    
    return result
}

// Example usage
let inputString = "Hello world from Swift"
let reversedString = reverseWords(in: inputString)
print(reversedString)

